assets/js/pages/HostDetailsPage/ProviderDetails.test.jsx

```diff
@@ -43,4 +45,80 @@ describe('Provider Details', () => {
       'Provider not recognized'
     );
   });
+
+  describe('AWS', () => {
+    const metadataLabels = [
+      {
+        key: 'account_id',
+        label: 'Account ID',
+      },
+      {
+        key: 'ami_id',
+        label: 'AMI ID',
+      },
+      // availability_zone is not displayed on its own but as part of the region
+      // {
+      //   key: 'availability_zone',
+      //   label: null,
+      // },
+      {
+        key: 'data_disk_number',
+        label: 'Data disk number',
+      },
+      {
+        key: 'instance_id',
+        label: 'Instance ID',
+      },
+      {
+        key: 'instance_type',
+        label: 'Instance type',
+      },
+      {
+        key: 'region',
+        label: 'Region',
+      },
+      {
+        key: 'vpc_id',
+        label: 'VPC ID',
+      },
+    ];
+
+    it('should render with AWS metadata', () => {
+      const awsMetadata = {
+        account_id: '123456',
+        ami_id: 'ami-67890',
+        availability_zone: 'eu-west-1a',
+        data_disk_number: 1,
+        instance_id: 'i-44444',
+        instance_type: 't3.micro',
+        region: 'eu-west-1',
+        vpc_id: 'vpc-99999',
+      };
+
+      render(
+        <ProviderDetails provider={AWS_PROVIDER} provider_data={awsMetadata} />
+      );
+
+      expect(screen.getByText('AWS')).toBeVisible();
+
+      metadataLabels.forEach(({ key, label }) => {
+        expect(screen.queryByText(label)).toBeVisible();
+
+        expect(screen.queryByText(label).nextSibling).toHaveTextContent(
+          awsMetadata[key],
+          { exact: false }
+        );
+      });
+    });
+
+    it('should render without AWS', () => {
+      render(<ProviderDetails provider={AWS_PROVIDER} provider_data={null} />);
+
+      expect(screen.getByText('AWS')).toBeVisible();
+
+      metadataLabels.forEach(({ _, label }) => {
+        expect(screen.queryByText(label)).toBeNull();
```

Comment:
nitpick: maybe `expect(screen.queryByText(label)).not.toBeInTheDocument()` would be more descriptive

Reply:
Here we go 535b7dd2f9c2f76b40d2e9b9f8270c0568ed5359